utils/replicaset/replicaset_test.go

```diff
@@ -778,3 +778,218 @@ func TestGetReplicaSetRevision(t *testing.T) {
 		assert.Equal(t, -1, GetReplicaSetRevision(ro, rs))
 	})
 }
+
+func TestGetRolloutAffinity(t *testing.T) {
+	ro := generateRollout("nginx")
+	assert.Nil(t, GetRolloutAffinity(ro))
+
+	ro.Spec.Strategy.BlueGreen = &v1alpha1.BlueGreenStrategy{
+		AntiAffinity: &v1alpha1.AntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1alpha1.RequiredDuringSchedulingIgnoredDuringExecution{},
+		},
+	}
+	assert.NotNil(t, GetRolloutAffinity(ro))
+
+	ro.Spec.Strategy.BlueGreen = nil
+	ro.Spec.Strategy.Canary = &v1alpha1.CanaryStrategy{
+		AntiAffinity: &v1alpha1.AntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: &v1alpha1.PreferredDuringSchedulingIgnoredDuringExecution{
+				Weight: 1,
+			},
+		},
+	}
+	assert.NotNil(t, GetRolloutAffinity(ro))
+}
+
+func TestGenerateReplicaSetAffinity(t *testing.T) {
```

Comment:
Creating subtests won't necessarily make it readable since I have to create Affinity objects to pass into the function for tests.

I added more detailed comments instead.